baybe/parameters/numerical.py

```diff
@@ -54,12 +56,14 @@ def _validate_tolerance(  # noqa: DOC101, DOC103
         Raises:
             ValueError: If the tolerance is not safe.
         """
-        # NOTE: computing all pairwise distances can be avoided if we ensure that the
-        #   values are ordered (which is currently not the case)
-        dists = pdist(np.asarray(self.values).reshape(-1, 1))
-        max_tol = dists.min() / 2.0
+        min_dist = np.diff(self.values).min()
+        if min_dist == (eps := np.nextafter(0, 1, dtype=DTypeFloatNumpy)):
+            raise NumericalUnderflowError(
+                f"The distance between any two parameter values must be at least "
```

Comment:
Hm, perhaps the case `tolerance=0.0` is special in the sense that we wouldn't even need to carry out the validation in the first place? Because the only requirement for `tolerance=0.0` is really that all values are distinct, which is however already asserted by the unique value validator. As soon as we have non-zero tolerance though, I run into the potential floating point underflow due to the `/ 2.0` division and couldn't quickly find a better alternative 🤔 So what do you suggest? Would it suffice to have a `if tolerance == 0.0` guard clause that skips validation in that case? Or do we need to polish the underflow logic?

Reply:
imo adding the special case to the valdiator will suffice at the moment

Reply:
done